presidio-image-anonymizer/tests/test_image_analyzer_engine.py

```diff
@@ -0,0 +1,166 @@
+    assert expected_exception_message == e.value.args[0]
+
+
+def test_given_valid_ocr_results_and_entities_then_map_entities_returns_correct_output(
+    get_ocr_analyzer_results, get_image_recognizerresult
+):
+    ocr_result, text, recogniser_result = get_ocr_analyzer_results
+    expected_result = get_image_recognizerresult
+    mapped_entities = ImageAnalyzerEngine.map_entities(
+        recogniser_result, ocr_result, text
+    )
+
+    assert len(expected_result) == len(mapped_entities)
+    assert expected_result == mapped_entities
+
+
+def test_given_empty_ocr_dict_and_entities_lists_then_map_entities_returns_empty_list(
+    get_ocr_analyzer_results,
+):
+    ocr_result, text, recogniser_result = get_ocr_analyzer_results
+    assert ImageAnalyzerEngine.map_entities([], {}, "") == []
+    assert ImageAnalyzerEngine.map_entities([], ocr_result, text) == []
+    assert ImageAnalyzerEngine.map_entities(recogniser_result, {}, "") == []
+
+
+def test_given_wrong_keys_in_ocr_result_dict_then_map_entities_returns_exception():
+    ocr_result = {"words": ["John"], "level": [0]}
+    expected_exception_message = "Key 'text' not found in dictionary"
+    with pytest.raises(KeyError) as e:
+        ImageAnalyzerEngine.map_entities([], ocr_result, "")
+    assert expected_exception_message == e.value.args[0]
+
+
+def test_given_repeated_entities_then_map_entities_returns_correct_number_of_bboxes(
+    get_ocr_analyzer_results,
+):
+    ocr_result, text, recogniser_result = get_ocr_analyzer_results
+
+    ocr_result["text"][1] = "Katie"
+    recogniser_result.append(RecognizerResult("PERSON", 1, 6, 0.85))
+    text = " Katie Interiors was created by Katie  Cromley."
+
+    assert (
+        len(ImageAnalyzerEngine.map_entities(recogniser_result, ocr_result, text)) == 3
+    )
+
+
+def test_given_word_has_entity_but_not_entity_then_map_entity_returns_correct_bboxes(
+    get_ocr_analyzer_results, get_image_recognizerresult
+):
+    ocr_result, text, recogniser_result = get_ocr_analyzer_results
+    ocr_result["text"][2] = "Katieiors"
+    text = " Homey Katieiors was created by Katie  Cromley."
+    expected_result = get_image_recognizerresult
+    mapped_entities = ImageAnalyzerEngine.map_entities(
+        recogniser_result, ocr_result, text
+    )
+
+    assert len(expected_result) == len(mapped_entities)
+    assert expected_result == mapped_entities
+
+
+def test_given_entity_with_multiple_words_then_map_entities_returns_correct_bboxees(
+    get_ocr_analyzer_results,
+):
+    ocr_result, text, recogniser_result = get_ocr_analyzer_results
+
+    # create new object for multiple words entities
+    recogniser_result = [RecognizerResult("PERSON", 32, 46, 0.85)]
+    expected_result = []
+    expected_result.append(
+        ImageRecognizerResult("PERSON", 32, 46, 0.85, 896, 64, 183, 40)
+    )
+    expected_result.append(
+        ImageRecognizerResult("PERSON", 32, 46, 0.85, 141, 134, 190, 50)
+    )

```

Comment:
expected_result = [ImageRecognizerResult("PERSON", 32, 46, 0.85, 896, 64, 183, 40), ImageRecognizerResult("PERSON", 32, 46, 0.85, 141, 134, 190, 50)] 
as well in the scenarios above, for immutability sake

Reply:
changed in all the tests.